### ISSUE TITLE
Aborting upload generates error

### ISSUE THREAD

**Issue by RavDanny** [VeliovGroup/Meteor-Files#398]
Hi
I'm using version 1.7.13.

I have set up my code to allow multiple file uploads using a ReactiveArray. Each file has a 'cancel' button which aborts that file's upload.

When aborting the upload, it triggers an on('end', function(error, fileObj)... event, returning a 408 Error (Can't continue upload, session expired. Start upload again.) and fileObj set to null. I'm assuming this is not normal behaviour.

Here is my code:
#### uploadForm.js
```
Template.uploadForm.onCreated(function () {
  this.currentUploads = new ReactiveArray();
});

Template.uploadForm.events({
  'change #fileInput': function (e, template) {
    var userId = Meteor.userId();
    if (e.currentTarget.files) {
      _.each(e.currentTarget.files, function(currentFile) {
          var upload = StoredFiles.insert({
            file: currentFile,
            streams: 'dynamic',
            chunkSize: 'dynamic',
            meta: {
              dirty: true
            }
          }, false);

          upload.on('start', function () {
            template.currentUploads.push(this);
            Session.set('uploading', true);
          });

          upload.on('end', function (error, fileObj) {
            if (error) {
              serverLog.error('Error during upload', currentFile, userId);
              alert('Error during upload: ' + error);
            } else {
              serverLog.info('File "' + fileObj.name + '" successfully uploaded', getDetails(fileObj) , userId);

              var fileRecord = {
                path: fileObj.path,
                name: fileObj.name,
                size: fileObj.size,
                key: fileObj._id,
                created: new Date(),
              };
              fileRecord.downloadable = document.getElementById('downloadable').checked;
            }

            template.currentUploads.remove(this);
            if (template.currentUploads.length == 0) {
              Session.set('uploading', false);
            }
          });

          upload.start();

        }
      });
    }
  },

  'click .cancel-upload': function(event, template) {
    event.preventDefault();
    var fileId = $(event.target).attr('id');
    var fileList = template.currentUploads.list();
    fileList[fileId].abort();
    return false;
  }
});
```
#### uploadForm.html
```
<template name="uploadForm">
  {{#if currentUploads}}
    <div>Uploading:</div>
    {{#each currentUploads}}
      <div>
        <b>{{file.name}}</b>:
        <span id="progress">{{progress.get}}%</span>
        <button id={{@index}} class="cancel-upload btn btn-xs">Cancel</button>
      </div>
    {{/each}}
  {{else}}
    <input id="fileInput" type="file" multiple />
  {{/if}}
</template>
```
Note: I initially also had the following .on('abort', ... ) function, intending that only it should run after the abort and not .on('end',... ), but I found both were called - it was called immediately when the cancel button was clicked, while the .on('end',...) function was called a second or so later.
```
          upload.on('abort', function (error, fileObj) {
            template.currentUploads.remove(this);
            if (template.currentUploads.length == 0) {
              Session.set('uploading', false);
            }
          });
```

Am I correct to assume this is also not expected behaviour?
If it is expected, I would like it to call just one function immediately after the abort. Is there a way to do this?

Thanks

**Comment by dr-dimitru** [comment]
Hello @RavDanny ,

This behavior caused by package asynchronous nature.

Whenever you call `abort` it terminates currently processed chunks and all further from being sent over the wire, and tells to the server to stop accepting that file's chunks. Although Client doesn't sending chunks anymore and Server doesn't accepts new one, there might be some chunks which was in the "middle" between Server and Client at the moment when `abort` was called, in this case Server will return `408` (*looks like I've picked wrong code*).

Anyways, all errors should be ignored after `abort` was called.

Let me know if I have explained this behavior well.
If you're expecting different behavior, your thoughts and wishes is very welcome [here](https://github.com/VeliovGroup/Meteor-Files/issues/309)
